### PR TITLE
S115 msft polish

### PR DESCRIPTION
**Pull request by eschultink**
### Fixes
  - clean up terraform script
  - fix grant module (need manual steps for Application permissions)
  - warning + config issues 

### Change implications

 - dependencies added/changed? **no**

### CODE REVIEW

File: infra/examples/aws-msft-365/README.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ running locally/interactively and our documentation will generally presume you'r
 approach.  If that doesn't suite your organization needs, the Terraform docs linked above provide
 alternatives, but this will require your own modification to the example configuration.
 
+If your target MSFT tenant (specified in `terraform.tfvars`) lacks an Azure subscriptiong (eg, is
```
Comment (aperez-worklytics) [inline infra/examples/aws-msft-365/README.md:21]:
```suggestion
If your target MSFT tenant (specified in `terraform.tfvars`) lacks an Azure subscription (eg, is
```

File: infra/modules/azuread-grant-all-users/main.tf [hunk 2]
```diff
@@ -10,25 +10,32 @@ terraform {
   }
 }
 
-data "azuread_application_published_app_ids" "well_known" {}
+# TODO: if grant can be made fully through API, do it here; until then, TODO file is best option
```
Comment (aperez-worklytics) [inline infra/modules/azuread-grant-all-users/main.tf:13]: True, that command seems to work only for delegated permissions...

To perform an [admin consent](https://docs.microsoft.com/en-us/cli/azure/ad/app/permission?view=azure-cli-latest#az-ad-app-permission-admin-consent):

```
az ad app permission admin-consent --id {appId}
```

To add a grant in the AAD application:

```
az ad app permission grant --id {appId} --api {permissionId} 
```

All permissions we have configured are application are *application* kind, grant admin will be the right one for these

**Review by aperez-worklytics — APPROVED** [review]
Just an example of commands in `az` to run the grants, in case you want to add it as part of the generated doc file.

File: infra/modules/azuread-grant-all-users/main.tf (hunk 2 above)
Comment (eschultink) [reply infra/modules/azuread-grant-all-users/main.tf:13]: weird thing is that it didn't give errors; the grant just had no effect

**Pull request merged**